src/contractTest/java/uk/gov/hmcts/reform/civil/provider/CivilCitizenUiProviderContractTest.java

```diff
@@ -0,0 +1,159 @@
+package uk.gov.hmcts.reform.civil.provider;
+
+import au.com.dius.pact.provider.junit5.PactVerificationInvocationContextProvider;
+import au.com.dius.pact.provider.junitsupport.Provider;
+import au.com.dius.pact.provider.junitsupport.State;
+import au.com.dius.pact.provider.junitsupport.loader.PactBroker;
+import au.com.dius.pact.provider.junitsupport.loader.PactBrokerConsumerVersionSelectors;
+import au.com.dius.pact.provider.junitsupport.IgnoreNoPactsToVerify;
+import au.com.dius.pact.provider.junitsupport.loader.SelectorBuilder;
+import au.com.dius.pact.provider.spring.junit5.MockMvcTestTarget;
+import au.com.dius.pact.provider.junit5.PactVerificationContext;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import uk.gov.hmcts.reform.civil.controllers.fees.FeesPaymentController;
+import uk.gov.hmcts.reform.civil.enums.FeeType;
+import uk.gov.hmcts.reform.civil.model.CardPaymentStatusResponse;
+import uk.gov.hmcts.reform.civil.service.FeesPaymentService;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@Provider("civil-service")
+@PactBroker(
+    url = "${PACT_BROKER_FULL_URL:http://localhost:80}",
+    providerBranch = "${pact.provider.branch}"
+)
+@IgnoreNoPactsToVerify
+class CivilCitizenUiProviderContractTest {
+
+    private static final String AUTH_HEADER = "Bearer some-access-token";
+    private static final String CASE_REFERENCE = "1234567890123456";
+    private static final String PAYMENT_REFERENCE = "RC-1701-0909-0602-0418";
+
+    private MockMvc mockMvc;
+
+    @Mock
+    private FeesPaymentService feesPaymentService;
+    private AutoCloseable mocks;
+
+    @PactBrokerConsumerVersionSelectors
+    public static SelectorBuilder consumerVersionSelectors() {
+        return new SelectorBuilder()
+            .matchingBranch()
+            .mainBranch()
+            .deployedOrReleased();
+    }
+
+    @BeforeEach
+    void beforeEach(PactVerificationContext context) {
+        System.setProperty("pact.verifier.publishResults", "true");
```

Comment:
The publishResults env var should not be hardcoded here or in all other provider tests and should be derived from the build.gradle (only publish verification results from the master branch)

Reply:
I was testing it locally and forgot to move it. Sorted